controllers/cloud.redhat.com/makers/maker.go

```diff
@@ -433,20 +434,19 @@ func (m *Maker) runProviders() (*config.AppConfig, error) {
 	c.MetricsPort = int(m.Env.Spec.Providers.Metrics.Port)
 	c.MetricsPath = m.Env.Spec.Providers.Metrics.Path
 
-	if err := objectstore.RunAppProvider(provider, &c, m.App); err != nil {
-		return &c, errors.Wrap("setupenv: getobjectstore", err)
-	}
-	if err := logging.RunAppProvider(provider, &c, m.App); err != nil {
-		return &c, errors.Wrap("setupenv: logging", err)
-	}
-	if err := kafka.RunAppProvider(provider, &c, m.App); err != nil {
-		return &c, errors.Wrap("setupenv: kafka", err)
-	}
-	if err := inmemorydb.RunAppProvider(provider, &c, m.App); err != nil {
-		return &c, errors.Wrap("setupenv: inmemorydb", err)
-	}
-	if err := database.RunAppProvider(provider, &c, m.App); err != nil {
-		return &c, errors.Wrap("setupenv: database", err)
+	sort.Sort(providers.ProvidersRegistration)
+
+	for _, provAcc := range providers.ProvidersRegistration {
+		if prov, err := provAcc.ProviderGet(&provider); err != nil {
+			return &c, errors.Wrap(fmt.Sprintf("getprov: %s", provAcc.Name), err)
+		} else if err == nil {
```

Comment:
why add the `if`? isn't `err == nil` the only other possibility if the original `if` checks `err != nil`?